adafruit_is31fl3731.py

```diff
@@ -1,13 +1,48 @@
-# CircuitPython compatible version of the micropython-adafruit-is31fl3731
-# Charlieplex LED driver:
-#  https://github.com/adafruit/micropython-adafruit-is31fl3731
-# Original micropython-adafruit-is31fl3731 author:
-#   Radomir Dopieralski
-# Port to CircuitPython API:
-#   Tony DiCola
+# The MIT License (MIT)
+#
+# Copyright (c) 2017 Toni DiCola
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+#pylint: disable-msg=line-too-long
+"""
+`adafruit_is31fl3731`
+====================================================
+
+CircuitPython driver for the IS31FL3731 charlieplex IC.
+
+This driver supports the following hardware:
+
+* `Adafruit 16x9 Charlieplexed PWM LED Matrix Driver - IS31FL3731 <https://www.adafruit.com/product/2946>`_
+* `Adafruit 15x7 CharliePlex LED Matrix Display FeatherWings <https://www.adafruit.com/product/2965>`_
+
+* Author(s): Toni DiCola
+"""
+#pylint: enable-msg=line-too-long
+
+# imports
 import math
 import time
+from micropython import const
 
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_IS31FL3731.git"
```

Comment:
You can break long strings just like in C:

```
__repo__ = ("https://github.com/adafruit/"
            "dafruit_CircuitPython_IS31FL3731.git")
```
and then you don't have to disable the line too long warning.

Reply:
pylint was complaining about the module description line length
```
* `Adafruit 16x9 Charlieplexed PWM LED Matrix Driver - IS31FL3731 <https://www.adafruit.com/product/2946>`_
* `Adafruit 15x7 CharliePlex LED Matrix Display FeatherWings <https://www.adafruit.com/product/2965>`_
```
I didn't know how to split a quoted doc string, i decided to allow long lines in this comment area.

Reply:
You can just put a newline in place of any space.